### ISSUE TITLE
Dynamic point parameter handling

### ISSUE THREAD

**Issue by nmellado** [poncateam/ponca#110]
Point parameters are specified statically by the 'MyPoint' class. Could be interesting to provide dynamic add/removal of the attribute fields.

**Comment by BDoignies** [comment]
@nmellado How would you do this? The only way I know is to have a special Point class that have a dynamic structure, something like std::map<std::string, std::variant<PROPERTIES>> (cuda? performances? API (how to add a property to a whole pointset)? force name for position,normal,...?). Otherwise, we are locked with templates... I also don't know if ponca should provide this type or if it is the user responsibility.

**Comment by nmellado** [comment]
yeah, I think that the way to go would be to have maps or equivalent. However it is not a priority right now, and alternatives can be used in downstream codes in the meantime.
